sdk/plus_queries.go

```diff
@@ -0,0 +1,99 @@
+package sdk
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/glifio/go-pools/abigen"
+	poolstypes "github.com/glifio/go-pools/types"
+)
+
+func (q *fevmQueries) PlusTokenIDFromRcpt(ctx context.Context, receipt *types.Receipt) (*big.Int, error) {
+	client, err := q.extern.ConnectEthClient()
+	if err != nil {
+		return nil, err
+	}
+	defer client.Close()
+
+	plusABI, err := abigen.PlusMetaData.GetAbi()
+	if err != nil {
+		return nil, err
+	}
+
+	plusFilterer, err := abigen.NewPlusFilterer(q.plus, client)
+	if err != nil {
+		return nil, err
+	}
+
+	var tokenID *big.Int
+
+	for _, l := range receipt.Logs {
+		event, err := plusABI.EventByID(l.Topics[0])
+		if err != nil {
+			return nil, err
+		}
+		if event.Name == "CardMinted" {
+			cardMintedEvent, err := plusFilterer.ParseCardMinted(*l)
+			if err != nil {
+				return nil, err
+			}
+			tokenID = cardMintedEvent.TokenId
+			break
+		}
+	}
+
+	return tokenID, nil
+}
+
+func (q *fevmQueries) PlusInfo(ctx context.Context, tokenID *big.Int) (*poolstypes.PlusInfo, error) {
+	client, err := q.extern.ConnectEthClient()
+	if err != nil {
+		return nil, err
+	}
+	defer client.Close()
+
+	plus, err := abigen.NewPlusCaller(q.plus, client)
+	if err != nil {
+		return nil, err
+	}
+
+	agentID, err := plus.TokenIdToAgentId(nil, tokenID)
+	if err != nil {
+		return nil, err
+	}
+
+	filCashbackEarned, err := plus.TokenIdToFilCashbackEarned(nil, tokenID)
+	if err != nil {
+		return nil, err
+	}
+
+	glfVaultBalance, err := plus.TokenIdToGlfVaultBalance(nil, tokenID)
+	if err != nil {
+		return nil, err
+	}
+
+	lastTierSwitchTimestamp, err := plus.TokenIdToLastTierSwitchTimestamp(nil, tokenID)
+	if err != nil {
+		return nil, err
+	}
+
+	personalCashBackPercent, err := plus.TokenIdToPersonalCashBackPercent(nil, tokenID)
+	if err != nil {
+		return nil, err
+	}
+
+	tier, err := plus.TokenIdToTier(nil, tokenID)
```

Comment:
the `nil` initial arguments can use the CallOpts from go-ethereum to pass the blockheight to query at

Reply:
I added a blockNumber argument